tutorials/01_neural_networks.ipynb

```diff
@@ -0,0 +1,957 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "reduced-twist",
+   "metadata": {},
+   "source": [
+    "# (Quantum) Neural Networks\n",
+    "\n",
+    "This notebook demonstrates the different generic (quantum) neural network implementations provided in Qiskit Machine Learning.\n",
+    "The networks are meant as application-anostic computational units that can be used in different use cases. \n",
+    "Depending on the application, a particluar type of network might more or less suitable.\n",
+    "In the following, the different available networks will be discussed in more detail:\n",
+    "\n",
+    "1. `NeuralNetwork`: The interface for neural networks.\n",
+    "2. `OpflowQNN`: A network based on the evaluation of quantum mechanical observables.\n",
+    "3. `TwoLayerQNN`: A special `OpflowQNN` implementation for convenience. \n",
+    "3. `CircuitQNN`: A network based on the samples resulting from measuring a quantum circuit."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "engaging-webster",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "from qiskit import Aer, QuantumCircuit\n",
```

Comment:
I guess this is a case where we really want to use Aer (for its snapshot capability i.e. using AerPauliExpectatio) and using BasicAer would not suffice.